app/src/main/java/com/artemzin/qualitymatters/network/HostSelectionInterceptor.java

```diff
@@ -0,0 +1,54 @@
+package com.artemzin.qualitymatters.network;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Such implementation allows us easily change base url in the integration and functional tests!
+ */
+public class HostSelectionInterceptor implements Interceptor {
+
+    @Nullable
+    private volatile String baseUrl;
+
+    @Override @NonNull
+    public Response intercept(@NonNull Chain chain) throws IOException {
+        Request request = chain.request();
+        final String baseUrl = this.baseUrl;
+
+        if (baseUrl != null) {
+            request = modifyRequest(request, baseUrl);
```

Comment:
Actually, we do not redirect all requests to the mock webserver.
By default `HostSelectionInterceptor` is dummy. It doesn't affect the current logic at all until `baseUrl` is set.
In fact, we added the interceptor that does nothing, though it has the property of dynamic change of the base url if there is a desire.